sql/core/src/main/scala/org/apache/spark/sql/execution/adaptive/OptimizeSkewedJoin.scala

```diff
@@ -250,6 +251,85 @@ case class OptimizeSkewedJoin(conf: SQLConf) extends Rule[SparkPlan] {
       }
   }
 
+  def optimizeSingleStageSkewJoin(plan: SparkPlan): SparkPlan = plan.transformUp {
```

Comment:
Is it possible to integrate this logic into the two-query-stage skew processing?
We can just treat the non-query-stage side as "canSplitXXXSide(...) = false", right?

Reply:
That's a good idea. I will have a try.